tests/readregs.rs

```diff
@@ -0,0 +1,52 @@
+//! This is a simple test to read registers from a child process.
+
+mod test_utils;
+
+#[cfg(target_os = "linux")]
+use headcrab::{target::LinuxTarget, target::UnixTarget};
+
+static BIN_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/tests/testees/hello");
+
+#[cfg(all(target_os = "linux", target_arch = "x86_64"))]
+#[test]
+fn read_memory() -> Result<(), Box<dyn std::error::Error>> {
+    test_utils::ensure_testees();
+
+    let target = LinuxTarget::launch(BIN_PATH)?;
+
+    let regs = target.read_regs()?;
+
+    // Assert that the register values match the expected initial values on Linux
+    assert_eq!(regs.r15, 0);
+    assert_eq!(regs.r14, 0);
+    assert_eq!(regs.r13, 0);
+    assert_eq!(regs.r12, 0);
+    assert_eq!(regs.rbp, 0);
+    assert_eq!(regs.rbx, 0);
+    assert_eq!(regs.r11, 0);
+    assert_eq!(regs.r10, 0);
+    assert_eq!(regs.r9, 0);
+    assert_eq!(regs.r8, 0);
+    assert_eq!(regs.rax, 0);
+    assert_eq!(regs.rcx, 0);
+    assert_eq!(regs.rdx, 0);
+    assert_eq!(regs.rsi, 0);
+    assert_eq!(regs.rdi, 0);
+    // sys_execve syscall: https://github.com/torvalds/linux/blob/f359287765c04711ff54fbd11645271d8e5ff763/arch/x86/entry/syscalls/syscall_64.tbl#L70
+    assert_eq!(regs.orig_rax, 59);
```

Comment:
This can probably be made into a const (with the same comment above):
```suggestion
    pub const SYSCALL_EXECVE: usize = 59;
    assert_eq!(regs.orig_rax, SYSCALL_EXECVE);
```

Reply:
I named it `X86_64_SYSCALL_EXECVE` as x86 uses a different number.